src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/factory/events/ApplicationSubmittedFactory.kt

```diff
@@ -0,0 +1,135 @@
+  private var sentenceLengthInMonths: Yielded<Int?> = { null }
+
+  fun withApplicationId(applicationId: UUID) = apply {
+    this.applicationId = { applicationId }
+  }
+
+  fun withApplicationUrl(applicationUrl: String) = apply {
+    this.applicationUrl = { applicationUrl }
+  }
+
+  fun withPersonReference(personReference: PersonReference) = apply {
+    this.personReference = { personReference }
+  }
+
+  fun withDeliusEventNumber(deliusEventNumber: String) = apply {
+    this.deliusEventNumber = { deliusEventNumber }
+  }
+
+  fun withMappaCategories(mappaCategories: String) = apply {
+    this.mappaCategories = { mappaCategories }
+  }
+
+  fun withOffenceDescription(offenceDescription: String) = apply {
+    this.offenceDescription = { offenceDescription }
+  }
+
+  fun withReleaseType(releaseType: String) = apply {
+    this.releaseType = { releaseType }
+  }
+
+  fun withAge(age: Int) = apply {
+    this.age = { age }
+  }
+
+  fun withGender(gender: ApplicationSubmitted.Gender) = apply {
+    this.gender = { gender }
+  }
+
+  fun withTargetLocation(targetLocation: String) = apply {
+    this.targetLocation = { targetLocation }
+  }
+
+  fun withSubmittedAt(submittedAt: OffsetDateTime) = apply {
+    this.submittedAt = { submittedAt }
+  }
+
+  fun withSubmittedByStaffMember(staffMember: StaffMember) = apply {
+    this.submittedByStaffMember = { staffMember }
+  }
+
+  fun withSubmittedByProbationArea(probationArea: ProbationArea) = apply {
+    this.submittedByProbationArea = { probationArea }
+  }
+
+  fun withSubmittedByCru(cru: Cru) = apply {
+    this.submittedByCru = { cru }
+  }
+
+  fun withSubmittedByTeam(team: Team) = apply {
+    this.submittedByTeam = { team }
+  }
+
+  fun withSubmittedByLdu(ldu: Ldu) = apply {
+    this.submittedByLdu = { ldu }
+  }
+
+  fun withSubmittedByRegion(region: Region) = apply {
+    this.submittedByRegion = { region }
+  }
+
+  fun withSentenceLengthInMonths(sentenceLengthInMonths: Int?) = apply {
+    this.sentenceLengthInMonths = { sentenceLengthInMonths }
+  }
+
+  override fun produce() = ApplicationSubmitted(
+    applicationId = this.applicationId(),
+    applicationUrl = this.applicationUrl(),
+    personReference = this.personReference(),
+    deliusEventNumber = this.deliusEventNumber(),
+    mappaCategories = this.mappaCategories(),
+    offenceDescription = this.offenceDescription(),
+    releaseType = this.releaseType(),
+    age = this.age(),
+    gender = this.gender(),
+    targetLocation = this.targetLocation(),
+    submittedAt = this.submittedAt(),
+    submittedBy = ApplicationSubmittedSubmittedBy(
+      staffMember = this.submittedByStaffMember(),
+      probationArea = this.submittedByProbationArea(),
+      cru = this.submittedByCru(),
+      team = this.submittedByTeam(),
+      ldu = this.submittedByLdu(),
+      region = this.submittedByRegion()
+    ),
+    sentenceLengthInMonths = this.sentenceLengthInMonths()
```

Comment:
when we reviewed `application.submitted` we identified a few changes but didn't actually write them up. I think what we agreed was:

- `mappaCategories`: this will comprise both mappa level and mappa category in a single string, e.g. `"CAT C3/LEVEL L2"`, so we could rename this to simply `mappa`?
- `offenceDescription`: we don't save the offence description, so let's return the `offenceId` instead and let the consumer lookup the human readable name
- `releaseType`: we noted that we'll be storing a type such as `["hdc", "licence", "pss", "rotl"]` rather than a string like `"Release on Temporary Licence (ROTL)"` So as long as we coerce that type list into a string (e.g. JSON) then we don't need to do anything beyond updating the "example" in OpenAPI spec? See [JSON schema in the UI](https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/d3816238499ed282fedcda46bdb0687e81690f1c/server/form-pages/apply/schema.json#L40)

Reply:
Good shout - have made these changes